Orchestrator.py

```diff
@@ -140,17 +143,137 @@ def receive(self,frame):
+        returnVal += [(x, y)]
+
+        # return current cell if still in same cell
+        if self._xy2hCell(startX, startY) == self._xy2hCell(stopX, stopY):
+            return returnVal
+
+        log.debug(f'next cell -> {(x, y)}')
+
+        if startX == stopX:
+
+            # vertical line, move down (increase y)
+            while True:
+                y += self.MINFEATURESIZE/2
+                returnVal += [(x, y)]
+                log.debug(f'next cell -> {(x,y)}')
+
+                if self._xy2hCell(x, y) == self._xy2hCell(stopX, stopY):
+                    break
+
+                maxLength = abs(math.ceil(stopY - startY))
+                log.debug(f'num cells {len(returnVal)} and maxlength = {abs(maxLength)}')
+                assert len(returnVal) <= (maxLength) * 4
+
+        else:
+
+            # move according to line equation y = mx + c
+            m  = (by - ay)/(bx - ax)
+            c  = startY - m*startX
+            log.debug(f'stop condition is {self._xy2hCell(stopX, stopY)}')
+
+            while True:
+                xmax  = (x + self.MINFEATURESIZE/2)
+                ynext = m*xmax + c
+                ymin  = y
+                ymax  = y + self.MINFEATURESIZE/2
+                log.debug(f'xmax,ymax {xmax}, {ymax}')
+                log.debug(f'ynext {ynext}')
+
+                if ynext < ymin:
+                    # move up
+                    y = y - self.MINFEATURESIZE/2
+                    returnVal += [(x, y)]
+                    log.debug(f'move up to -> {(x, y)}')
+
+                elif ynext > ymax:
+                    # move down
+                    y = y + self.MINFEATURESIZE/2
+                    returnVal += [(x, y)]
+                    log.debug(f'move down to -> {(x, y)}')
+
+                elif ymin < ynext < ymax:
+                    # move right
+                    x = x + self.MINFEATURESIZE/2
+                    returnVal += [(x, y)]
+                    log.debug(f'move right to -> {(x, y)}')
+
+                elif (ynext == ymin or ynext == ymax):
+                    # move diagonally if trajectory isnt on cell border
+                    if stopY != startY:
+                        y = y + self.MINFEATURESIZE / 2
+                    x = x + self.MINFEATURESIZE/2
+
+                    returnVal += [(x, y)]
+                    log.debug(f'move diagonally to -> {(x, y)}')
+
+                if self._xy2hCell(x, y) == self._xy2hCell(stopX, stopY):
+                    break
+
+                maxLength = abs(math.ceil(stopX - startX))+abs(math.ceil(stopY-startY))
+                log.debug(f'num cells {len(returnVal)} and maxlength = {abs(maxLength)}')
+                assert len(returnVal) <= (maxLength)*4
+
+
+        log.debug(f'new cells {returnVal}')
+        return returnVal
+
+    def _xy2hCell(self, x, y):
+
+        # convert x,y coordinates to cell (top left coordinate)
+        xsteps = int(math.floor((x-self.initX)/ (self.MINFEATURESIZE/2)))
```

Comment:
I don't understand why you use `initX` and `initY`. Please add a comment at the top of the function with an example input and output